### PR TITLE
Cert-manager-app fixes

### PR DESCRIPTION
**Pull request by ssyno**
This PR:
resolves [#28598](https://github.com/giantswarm/giantswarm/issues/28598)
- fixes issues around cert-manager-app
  - [x] IngressClass nginx has to exist for the acme-solvers to work
  - [x] ClusterIssuers exist after job completes
  - [x] NetworkPolicies & CiliumNetworkPolicies for giantswarm namespace

### Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/cert-manager/charts/cert-manager-giantswarm-clusterissuer/values.yaml [hunk 1]
```diff
@@ -46,4 +46,4 @@ acme:
       secretAccessKey: ""
   http01:
     enabled: true
-    ingressClassName: nginx
+    ingressClass: ""
```
Comment (anvddriesch) [inline helm/cert-manager/charts/cert-manager-giantswarm-clusterissuer/values.yaml:49]: For my intertest, what will be the outcome of removing the default?
Will it just result the solver ingresses to have no class or will they have the class of the original ingress? Could there be a negative outcome for cases in which other issuers/classes are defined?

**Review by vvondruska — APPROVED** [review]

File: helm/cert-manager/charts/cert-manager-giantswarm-clusterissuer/values.yaml (hunk 1 above)
Comment (ssyno) [reply helm/cert-manager/charts/cert-manager-giantswarm-clusterissuer/values.yaml]: I came to the conclusion that this change to "" or default doesn't makes sense for now, as there may be no ingress controllers deployed by the time cert-manager is deployed

**Review by calvix — APPROVED** [review]

File: helm/cert-manager/values.yaml [hunk 2]
```diff
@@ -787,6 +787,6 @@ startupapicheck:
   enableServiceLinks: false
 
 ciliumNetworkPolicy:
-  enabled: false
+  enabled: true
```
Comment (kopiczko) [inline helm/cert-manager/values.yaml:790]: We shouldn't enable it by default.

File: helm/cert-manager/charts/cert-manager-giantswarm-clusterissuer/templates/acme-solvers-networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    {{- include "issuerLabels" . | nindent 4 }}
+  annotations:
+    {{- include "issuerAnnotations" . | nindent 4 }}
+spec:
+  podSelector:
+    matchLabels:
+      acme.cert-manager.io/http01-solver: "true"
+  policyTypes:
+  - Ingress
+  egress:
```
Comment (kopiczko) [inline helm/cert-manager/charts/cert-manager-giantswarm-clusterissuer/templates/acme-solvers-networkpolicy.yaml:15]:
```suggestion
  ingress:
```

File: helm/cert-manager/charts/cert-manager-giantswarm-netpol/templates/acme-solvers-ciliumnetworkpolicy.yaml
Comment (kopiczko) [inline helm/cert-manager/charts/cert-manager-giantswarm-netpol/templates/acme-solvers-ciliumnetworkpolicy.yaml:1]: We should keep only one policy type

**Review by kopiczko — APPROVED** [review]

**Comment by T-Kukawka** [comment]
can we merge and release this? 🙏

**Pull request merged**